integration/integration_test.go

```diff
@@ -0,0 +1,145 @@
+package integration
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/CityOfZion/neo-go/pkg/core"
+	"github.com/CityOfZion/neo-go/pkg/core/transaction"
+	"github.com/CityOfZion/neo-go/pkg/io"
+	"github.com/docker/go-connections/nat"
+	"github.com/stretchr/testify/require"
+)
+
+// Main steps for testing are:
+// - prepare docker image with neo-go node or c# node
+// - start privatenet
+// - start docker nodeContainerReq
+// - create RPC client
+// - generate input for RPC client
+// - start sending txes to the node
+// - measure how much TX could be sent
+
+const (
+	// RpcPort for RPC calls to test nodes.
+	RpcPort = nat.Port("20331")
+	// NumberOfTx number of transactions to be generated.
+	NumberOfTx = 3000
+	// TimeoutInSeconds time in seconds to wait for test to finish.
+	TimeoutInSeconds = 500
+)
+
+// SendTXResponse struct for testing.
+type SendTXResponse struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  bool   `json:"result"`
+	ID      int    `json:"id"`
+}
+
+// GetBlockCountResponse struct for testing.
+type GetBlockCountResponse struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  int    `json:"result"`
+	ID      int    `json:"id"`
+}
+
+// GetBlockResponse struct for testing.
+type GetBlockResponse struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  string `json:"result"`
+	ID      int    `json:"id"`
+}
+
+func TestIntegration(t *testing.T) {
+	callNode(t, RpcPort)
+}
+
+// makes a new RPC client and calls node by RPC.
+func callNode(t *testing.T, port nat.Port) {
+	client := NewRPCClient(port)
+	data := getTransactionsForIntegration(t, NumberOfTx)
+	startTime := uint32(time.Now().UTC().Unix())
+	// -1 since getBlockCount returns block count including first zero block
+	startBlockIndex := getBlockCount(t, client) - 1
+	log.Printf("Started test from block = %v at unix time = %v", startBlockIndex, startTime)
+	for _, tx := range data {
+		body := client.SendTX(tx, t)
+		var resp SendTXResponse
+		err := json.Unmarshal(body, &resp)
+		require.NoError(t, err)
+	}
+	log.Println("All transactions were sent")
+
+	lastBlockTimestamp := startTime
+	searchAllTX(t, client, startBlockIndex, data, &lastBlockTimestamp)
+
+	log.Printf("Sent %v transactions in %v seconds", NumberOfTx, lastBlockTimestamp-startTime)
```

Comment:
I think you can basically throw away the first and the last blocks and print per-block statistics like `block X generated in Y seconds with Z transactions (Z/Y TPS)`. Comparing things to `startTime` doesn't seem correct to me because you have the first block that's gonna contain some random number of transactions (depending upon where in the block creation cycle you're to start testing) and you have the last block that just drains the mempool to zero irrespective of the number of transactions it contains (1 or 1000) in its 15 seconds. The interesting blocks are all in the middle.

Reply:
hm don't get what you mean.
If you create just 500 TX you have only two blocks in test. First - when test started and second - with accepted 500 TX. There are no blocks in the middle.
And it's not clear how to calculate time diff then.

Reply:
To me that just means that 500 TXes is a bad test that tells nothing real about TPS value.

Reply:
as for me calculating TPS is weird too since it doesn't give any business value but it's not answering the previous question